crds_data.ipynb

```diff
@@ -533,6 +533,95 @@
     "df.head()\n"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Apply logbook flags"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def read_logbook(path):\n",
+    "    \"\"\"\n",
+    "    Read events in the logbook file (.csv) \n",
+    "    Drop empty lines.\n",
+    "    \n",
+    "    Dataframe columns name: Event | Start_Event | End_Event | Flags\n",
+    "    \n",
+    "    Parameters: path of .csv logbook\n",
+    "    Returns: dataframe with the logbook data\n",
+    "    \"\"\"\n",
+    "    lines = open(path_logbook).readlines()\n",
+    "    lines_skip = lines.index('Event,\"Start Event\\n')\n",
+    "    df = pd.read_csv(path, sep=',',  skiprows=lines_skip)\n",
+    "    df = df.rename(columns={\"Start Event\\ndd/mm/aaaa hh:mm\": \"Start_Event\",\n",
+    "                           \"End Event\\ndd/mm/aaaa hh:mm\": \"End_Event\"})\n",
+    "    df = df.dropna(how='all')\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def drop_invalid_logbook(df):\n",
+    "    \"\"\"\n",
+    "    Drop invalid datatime in logbook dataframe.\n",
+    "    Index are reseted.\n",
+    "    \n",
+    "    Parameters: logbook dataframe\n",
+    "    Returns: logbook dataframe with valid datatime\n",
+    "    \"\"\"\n",
+    "    for line in range(1,len(df['Event'])+1):\n",
+    "        try:\n",
+    "            df['Start_Event'][line] = datetime.strptime(df['Start_Event'][line], '%d/%m/%Y %H:%M')\n",
+    "            df['End_Event'][line] = datetime.strptime(df['End_Event'][line], '%d/%m/%Y %H:%M')\n",
+    "        except:\n",
+    "            df = df.drop(index=line)\n",
+    "    df = df.reset_index()\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def insert_flag(df, df_logbook):\n",
+    "    \"\"\"\n",
+    "    Insert column flag for manual control quality.\n",
+    "    \n",
+    "    Parameters: data dataframe (df), logbook dataframe (df_logbook)\n",
+    "    Returns: data dataframe (df) with flags\n",
+    "    \"\"\"\n",
+    "    df['FLAGS'] = np.nan\n",
+    "    lines_logbook = range(df_logbook['Start_Event'].shape[0])\n",
+    "    for i in lines_logbook:\n",
+    "        df.loc[(df.index >= df_logbook['Start_Event'][i]) & \n",
+    "                   (df.index <= df_logbook['End_Event'][i]),['FLAGS']] = df_logbook['Flags'][i]\n",
```

Comment:
We have to consider cases where more than one flag can be applied in overlapping periods

Reply:
Here, I dropped repeated characters. e.g.:
KC + KI = KCI
KC + NS = KCNS